examples/connext_dds/builtin_topics/c++11/msg_subscriber.cxx

```diff
@@ -62,69 +65,80 @@ void subscriber_main(
         std::cout << "error, participant user_data exceeds resource limits"
                   << std::endl;
     } else {
-        participant_qos << UserData(
-                ByteSeq(participant_auth.begin(), participant_auth.end()));
+        participant_qos << dds::core::policy::UserData(dds::core::ByteSeq(
+                participant_auth.begin(),
+                participant_auth.end()));
     }
 
     // Create a DomainParticipant.
-    DomainParticipant participant(domain_id, participant_qos);
+    dds::domain::DomainParticipant participant(domain_id, participant_qos);
 
     // Participant is disabled by default in USER_QOS_PROFILES. We enable it now
     participant.enable();
 
     // Create a Topic -- and automatically register the type.
-    Topic<msg> topic(participant, "Example msg");
+    dds::topic::Topic<msg> topic(participant, "Example msg");
+
+    // Create a DataReader with default QoS
+    // Create a Subscriber and DataReader with default Qos
+    dds::sub::Subscriber subscriber(participant);
+    dds::sub::DataReader<msg> reader(subscriber, topic);
 
-    // Create a DataReader
-    DataReader<msg> reader(Subscriber(participant), topic);
+    // WaitSet will be woken when the attached condition is triggered
+    dds::core::cond::WaitSet waitset;
 
-    // Create a data reader listener using ListenerBinder, a RAII utility that
-    // will take care of reseting it from the reader and deleting it.
-    ListenerBinder<DataReader<msg>> scoped_listener = bind_and_manage_listener(
+    // Create a ReadCondition for any data on this reader, and add to WaitSet
+    unsigned int samples_read = 0;
+    dds::sub::cond::ReadCondition read_condition(
             reader,
-            new MsgListener,
-            dds::core::status::StatusMask::data_available());
+            dds::sub::status::DataState::any(),
+            [reader, &samples_read]() {
+                // If we wake up, process data
```

Comment:
This comment is everywhere and it's both inaccurate and useless. Is it still being generated with rtiddgen? I gave the same feedback.